core/src/main/java/org/elasticsearch/index/query/TermsSetQueryBuilder.java

```diff
@@ -290,7 +291,6 @@ public boolean needsScores() {
 
         @Override
         public int hashCode() {
-            // CoveringQuery with this field value source cannot be cachable
```

Comment:
Same here, probably not important but we can use the script for hashCode and equals ?